### ISSUE TITLE
Disable Reports to Kong by default

### ISSUE THREAD

**Issue by austince** [kumahq/kuma#3050]
### Summary

I've just stumbled upon the "reports" feature, which looks to send usage data to `kong-hf.konghq.com`. It also seems like this is enabled by default: https://github.com/kumahq/kuma/blob/f07e2d488205d1d883cbf95f803b705f7727cf92/pkg/config/app/kuma-cp/config.go#L193-L195 

It seems to publish:
- \# of services (internal + external)
- \# of meshes
- \# of zones
- cluster id 

In line with disabling the RUM features in the GUI (https://github.com/kumahq/kuma-gui/pull/210), I think we should also disable this by default, if not remove it entirely or at least publish the data in a vendor-neutral place.

I can't find any mention of this feature on the website, bar one ambiguous `reports` key in the configuration. Am I missing it somewhere / missing some other bit of context?

cc @jakubdyszkiewicz @lobkovilya @lahabana

**Comment by austince** [comment]
It seems like this also sends:

* hostname
* backend type (postgres, k8s)
* version information

**Comment by austince** [comment]
The CNCF shared some good guidance in this Service Desk ticket: https://cncfservicedesk.atlassian.net/servicedesk/customer/portal/1/CNCFSD-982
Mainly, what seems to apply here is:
* https://www.linuxfoundation.org/telemetry-data-policy/

Some key questions it asks, which are relevant in our situation, are:

> Does the Telemetry Data lead to the ability to track or uniquely identify the user? Even if it doesn’t, does the Telemetry Data otherwise include some form of personal information that is subject to laws and regulations, or even just that the user doesn’t realize is being shared?

The answer seems to be yes to all, given the hostname in particular.

> Does the Telemetry Data result in any potentially business-sensitive information being sent to the project community? Does a business realize that the open-source software is sharing this data? Even if a staff member clicked to consent, were they authorized to enable data sharing on behalf of their employer?

Again, I believe this is the case here, given the hostname and perhaps the IPs.

> Does the software ensure that all relevant users and installers of the software are aware of the Telemetry Data collection, before it is enabled? Is it opt-out or opt-in? Can notices or consents be inadvertently bypassed when the software is installed through automated means?

Unless I've missed something, I don't think consent is ever given (as it is opt-out and undocumented). 

> Does the “phone home” functionality open up any inadvertent security vulnerabilities? Could those vulnerabilities be present even for users who refuse to enable Telemetry Data?

I'm no security expert, but it doesn't seem to be too insecure, though the data appears to be sent in plaintext over a TCP connection. That's probably not great.

https://github.com/kumahq/kuma/blob/8f1f7ff4d677cb9b1131e0088b7dd4d644a9924d/pkg/core/runtime/reports/reports.go#L161-L162

The current policy:
> 
> By default, projects of the Linux Foundation should not collect Telemetry Data from users of open source software that is distributed on behalf of the project.
> 
> If a project community desires to collect Telemetry Data, it must first coordinate with members of the legal team of the Linux Foundation to undergo a detailed review of the proposed Telemetry Data and collection mechanism.

Has this happened before I came onboard?

**Comment by jpeach** [comment]
Either someone should do the legwork to get this telemetry approved in the CNCF process (based on the existing spinnaker approval, I expect it would be OK), or this should be removed.

**Comment by austince** [comment]
We talked about this during the [2021-11-10 community call](https://docs.google.com/document/d/1HgnK3wJIEg8uFlivdrhrPZYWTpElWWu3mhFDXj-bMWQ/edit#)

* There are discussions between Kong and CNCF currently about telemetry collection but are not public.
* The data collection will continue as is until those discussions reach a conclusion.
* The security vulnerabilities noted here will not be addressed until the above is resolved. 

As I stated on the call, I believe this approach will continue to harm the community.
